src/platform_impl/linux/mod.rs

```diff
@@ -1,12 +1,27 @@
-#![cfg(any(target_os = "linux", target_os = "dragonfly", target_os = "freebsd", target_os = "netbsd", target_os = "openbsd"))]
-
-use std::{collections::VecDeque, env, ffi::CStr, fmt, mem::MaybeUninit, os::raw::*, sync::Arc};
-
+#![cfg(any(
+    target_os = "linux",
+    target_os = "dragonfly",
+    target_os = "freebsd",
+    target_os = "netbsd",
+    target_os = "openbsd"
+))]
+
+#[cfg(all(not(feature = "x11"), not(feature = "wayland")))]
+compile_error!("Please select a feature to build for unix: `x11`, `wayland`");
```

Comment:
Please, do that in build.rs where it should belong, otherwise it's hard to notice such errors, since a lot of things errors out in the same time + it's more clear, when it's in build.rs

Reply:
I added a build.rs with that check, and also the check for the web features